services/tables/src/main/java/com/linkedin/openhouse/tables/toggle/FeatureToggleAspect.java

```diff
@@ -17,10 +24,41 @@
 @Aspect
 @Component
 @Slf4j
-public class FeatureGateAspect {
+public class FeatureToggleAspect {
 
   @Autowired private TableFeatureToggle tableFeatureToggle;
 
+  @Autowired private TblPropsToggleRegistry tblPropsToggleRegistry;
+
+  @Around("@annotation(tblPropsEnabler)")
+  public boolean checkTblPropEnabled(
+      ProceedingJoinPoint proceedingJoinPoint, TblPropsEnabler tblPropsEnabler) throws Throwable {
+    if (((MethodSignature) proceedingJoinPoint.getSignature()).getReturnType() == boolean.class
+        && proceedingJoinPoint.getArgs()[1] instanceof TableDto
+        && proceedingJoinPoint.getArgs()[0] instanceof String) {
+      TableDto tableDto = (TableDto) proceedingJoinPoint.getArgs()[1];
+      String key = (String) proceedingJoinPoint.getArgs()[0];
+
+      Optional<String> feature = tblPropsToggleRegistry.obtainFeatureByKey(key);
+      if (!feature.isPresent()) {
```

Comment:
nit: can we use map / orElseGet for the optional feature so that the reader knows that the code below this condition is only executed once the feature is known to be present

Reply:
addressed.